### PR TITLE
[Spot] Add Azure dependency installation on SpotController.

### PR DESCRIPTION
**Pull request by cblmemo**
Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
    - [ ] `sky spot launch --cloud azure --cpus 2 whoami`
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

**Review by concretevitamin — APPROVED** [review]
LGTM

**Review by Michaelvll — COMMENTED** [review]
I am a bit hestitating for installing azure dependencies on the spot controller, as the installation of the dependency takes significant amount of time. Is it possible we check if there is a credential file uploaded to the controller before we install a depdency of a cloud?

**Comment by cblmemo** [comment]
> I am a bit hestitating for installing azure dependencies on the spot controller, as the installation of the dependency takes significant amount of time. Is it possible we check if there is a credential file uploaded to the controller before we install a depdency of a cloud?

Make sense! Added. Thanks!

File: sky/utils/controller_utils.py [hunk 1]
```diff
@@ -151,9 +152,9 @@ def _get_cloud_dependencies_installation_commands(
         # dependencies for sky serve controller.
         commands.append('pip list | grep oci > /dev/null 2>&1 || '
                         'pip install oci > /dev/null 2>&1')
-    else:
-        # We do not install azure dependencies for spot controller since our
-        # subscription does not support spot instances.
+    # TODO(tian): Make dependency installation command a method of cloud class
+    # and get all installation command for enabled clouds.
+    if clouds.Azure() in global_user_state.get_enabled_clouds():
```
Comment (Michaelvll) [inline sky/utils/controller_utils.py:157]: won't this always be `False? clouds.Azure() is a different object then the one in `get_enabled_clouds()`. Check out the function: https://github.com/skypilot-org/skypilot/blob/1bb59fd25b7b2de9ccb65c52c42a356145aaddb7/sky/optimizer.py#L1088-L1091

Comment (cblmemo) [reply sky/utils/controller_utils.py]: Make sense! Thanks. Fixed!

**Review by Michaelvll — APPROVED** [review]
Thanks for the update @cblmemo! LGTM.

**Pull request merged**